core/services/chainlink/relayer_factory.go

```diff
@@ -84,43 +85,64 @@ func (r *RelayerFactory) NewSolana(ks keystore.Solana, chainCfgs solana.SolanaCo
 		signer  = &keystore.SolanaSigner{Solana: ks}
 	)
 
+	unique := make(map[string]struct{})
 	// create one relayer per chain id
 	for _, chainCfg := range chainCfgs {
+
 		relayId := relay.ID{Network: relay.Solana, ChainID: relay.ChainID(*chainCfg.ChainID)}
+		_, alreadyExists := unique[relayId.Name()]
+		if alreadyExists {
+			return nil, fmt.Errorf("duplicate chain definitions for %s", relayId.Name())
+		}
+		unique[relayId.Name()] = struct{}{}
+
+		// skip disabled chains from further processing
+		if !chainCfg.IsEnabled() {
+			solLggr.Warnw("Skipping disabled chain", "id", chainCfg.ChainID)
+			continue
+		}
+
 		// all the lower level APIs expect chainsets. create a single valued set per id
 		singleChainCfg := solana.SolanaConfigs{chainCfg}
 
 		if cmdName := env.SolanaPluginCmd.Get(); cmdName != "" {
 
 			// setup the solana relayer to be a LOOP
-			tomls, err := toml.Marshal(struct {
-				Solana solana.SolanaConfigs
-			}{Solana: singleChainCfg})
+			cfgTOML, err := toml.Marshal(struct {
+				Solana solana.SolanaConfig
+			}{Solana: *chainCfg})
+
 			if err != nil {
 				return nil, fmt.Errorf("failed to marshal Solana configs: %w", err)
 			}
 
 			solCmdFn, err := plugins.NewCmdFactory(r.Register, plugins.CmdConfig{
-				ID:  solLggr.Name(),
+				ID:  relayId.Name(),
 				Cmd: cmdName,
 			})
 			if err != nil {
 				return nil, fmt.Errorf("failed to create Solana LOOP command: %w", err)
 			}
-			solanaRelayers[relayId] = loop.NewRelayerService(solLggr, r.GRPCOpts, solCmdFn, string(tomls), signer)
+			// TODO change the solana loop toml deserializer so we can get rid of the singleChainCfg
```

Comment:
all todos should have a ticket linked otherwise they tend to fall into the dustbin of history